File: job-service/src/main/java/com/pbl6/jobservice/entity/Application.java
package com.pbl6.jobservice.entity;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;
import java.util.UUID;
@Entity
@Table(name = "applications")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Application extends Base{

    @Id
    @GeneratedValue
    @Column(columnDefinition = "uuid")
    UUID applicationId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_id", nullable = false)
    Job job;

    @Column(nullable = false, columnDefinition = "uuid")
    UUID applicantId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    Status status = Status.SUBMITTED;

    @Column(columnDefinition = "TEXT")
    String notes;

    @Column(nullable = false)
    LocalDateTime appliedDate;

    String cvFileUrl;

    public enum Status {
        SUBMITTED, REVIEWED, INTERVIEW, HIRED, REJECTED
    }
}

File: lib/features/user/jobs/domain/entities/application.dart
enum ApplicationStatus { pending, accepted, rejected }

class Application {
  final String id;
  final String jobId; // FK to Job
  final String applicantId; // FK to User
  final ApplicationStatus status;
  final String notes;
  final String cvFileUrl;
  final DateTime appliedDate;

  const Application({
    required this.id,
    required this.jobId,
    required this.applicantId,
    required this.status,
    required this.notes,
    required this.cvFileUrl,
    required this.appliedDate,
  });

  factory Application.fromJson(Map<String, dynamic> json) {
    return Application(
      id: json['applicationId'] ?? '',
      jobId: json['jobId'] ?? '',
      applicantId: json['applicantId'] ?? '',
      status: _parseStatus(json['status']),
      notes: json['notes'] ?? '',
      cvFileUrl: json['cvFileUrl'] ?? '',
      appliedDate: DateTime.tryParse(json['appliedDate'] ?? '') ??
          DateTime.fromMillisecondsSinceEpoch(0),
    );
  }

  static ApplicationStatus _parseStatus(String? status) {
    switch (status?.toLowerCase()) {
      case 'accepted':
        return ApplicationStatus.accepted;
      case 'rejected':
        return ApplicationStatus.rejected;
      default:
        return ApplicationStatus.pending;
    }
  }
}
